latch_cli/utils.py

```diff
@@ -115,27 +115,34 @@ def with_si_suffix(num, suffix="B", styled=False):
 def hash_directory(dir_path: Path) -> str:
     m = hashlib.new("sha256")
     m.update(current_workspace().encode("utf-8"))
-    for containing_path, dirnames, fnames in os.walk(dir_path):
+    exclude = None
+    ignore_file = dir_path / ".dockerignore"
+    if os.path.exists(ignore_file):
+        with open(ignore_file) as f:
+            exclude = list(
+                filter(
+                    lambda x: x != "" and x[0] != "#",
+                    [l.strip() for l in f.read().splitlines()],
+                )
+            )
+    for item in exclude_paths(dir_path, exclude):
```

Comment:
by getting rid of the `fnames.sort()` etc, you're relying solely on the order of paths returned by `exclude_paths` to determine the version hash value - is the order of paths always the same given the same directory?

Reply:
true